routes/login/login.controller.js

```diff
@@ -212,21 +215,24 @@ const postDateOfBirth = async (req, res, next) => {
   if (!errors.isEmpty()) {
     let errObj = errorArray2ErrorObject(errors)
 
-    /*
-    We don't want to show the "birthdate doesn't match" error if there
-    are other errors, because it is obvious the birthdate doesn't match if
-    you enter a month of 99.
-
-    If exists more than 1 error, and the match error exists, delete the match error
-    */
-    if (
-      Object.keys(errObj).length > 1 &&
-      errObj.dobDay &&
-      errObj.dobDay.msg === 'errors.login.dateOfBirth.match'
-    ) {
-      delete errObj.dobDay
+    return res.status(422).render('login/dateOfBirth', {
+      prevRoute: getPreviousRoute(req),
+      data: req.session,
+      body,
+      errors: errObj,
+    })
+  }
+  req.session.personal.dateOfBirth = _toISOFormat(req.body)
+
+  // check access code + SIN + DoB
+  // if session doesn't have a sin, throw error
+  if(!req.session.personal.sin) {
+    // error no sin
+    let errObj = {
+      sin: {
+        msg: 'errors.login.matchingSIN',
+      },
```

Comment:
Let's simplify this. 

Since we're not using the "matching DoB login" anymore, we can add [the `checkErrors` call](https://github.com/cds-snc/cra-claim-tax-benefits/blob/master/routes/login/login.controller.js#L49) as one middleware, and then in our second middleware we can make sure they all add up.

If we we find an error (sin is missing, SIN is wrong, Dob is wrong, etc), we should redirect back to the `/sin` page with an error message.  

As I am writing this, I am realizing that that would mean the url would be wrong, so maybe we have to check the session in the SIN post function.